static/js/compose_validate.js

```diff
@@ -654,6 +655,99 @@ function validate_private_message() {
     return true;
 }
 
+export function check_send_permissions() {
+    const $send_button = $("#compose-send-button");
+    const $send_button_container = $(".compose_right_float_container")[0];
+    const tippy_instance = $send_button_container._tippy || tippy($send_button_container);
+    const msg_type = compose_state.get_message_type();
+
+    if (msg_type === "private" && page_params.realm_private_message_policy === 2) {
+        const recipients = compose_pm_pill.get_user_ids();
+        const non_bot_users = recipients.filter(
+            (user_id) => !people.get_by_user_id(user_id).is_bot,
+        );
+
+        if (non_bot_users.length > 0) {
+            tippy_instance.setContent(
+                $t_html({
+                    defaultMessage:
+                        "You are not allowed to send private messages in this organization.",
+                }),
+            );
+            $send_button.prop("disabled", true);
+            return false;
+        }
+    }
+
+    if (msg_type === "stream") {
+        const stream_name = compose_state.stream_name();
+        const sub = stream_data.get_sub(stream_name);
+
+        if (!can_send_to_stream(sub)) {
+            if (sub) {
+                tippy_instance.setContent(
+                    $t_html(
+                        {
+                            defaultMessage:
+                                "You are not allowed to send messages to #{stream_name}.",
+                        },
+                        {stream_name},
+                    ),
+                );
+            } else {
+                tippy_instance.destroy();
```

Comment:
Can we instead have the tooltip saying stream name is invalid?

Reply:
Could you provide more details about the question/proposal here? Thanks!

Reply:
I suggested above that we should disable the button when stream name in the stream recipient input is invalid. So, I was suggesting here to have a tooltip in that case mentioning the stream name is invalid.

Reply:
Ah, sounds good! I think in the future, we'll use a dropdown for the stream rather than a free-text field (e.g. as we do in the move message UI), so if it's significant extra work, it's not worth worrying about here.

Reply:
I added this in the most recent commit, it wasn't too difficult to implement. I based the message on the existing error message in the same file